devito/finite_differences/coefficients.py

```diff
@@ -217,27 +227,38 @@ def generate_subs(i):
         return rules
 
 
+def extract_dim(dim):
+    if isinstance(dim, sympy.Add):
+        # Staggered function: need to extract dimension from dim
+        return list(dim.as_coefficients_dict())[-1]
```

Comment:
Is it always guarantied it's the last one?

Reply:
Not 100% sure, I assume it would be, but need to double check. Could potentially take both and check which one is a Dimension instead?

Reply:
Think of it, all you want is to know whether it's a time dimension or not right?

so you could have something like
```
def is_time(function, dim):
   any (getattr(i, 'is_Time', False) for i in dim.args)
```
and same for space (can inline in if below)

Reply:
In a couple of other places further down, I need the dimension on its own (without the offset). Would be nice to have a tidier/more reliable to get the dimension from the Add though